dimod/reference/samplers/exact_solver.py

```diff
@@ -73,30 +73,28 @@ def __init__(self):
         self.properties = {}
         self.parameters = {}
 
-    def sample(self, bqm, **kwargs):
+    def sample(self, bqm, **kwargs) -> SampleSet:
```

Comment:
If we're going to be touching other classes/methods to add annotations, we should also add the input types. E.g. `bqm: BinaryQuadraticModel`. You can avoid the circular imports that are likely to occur with
```python
if typing.TYPE_CHECKING:
    from dimod import BinaryQuadraticModel
```
see https://stackoverflow.com/questions/39740632/python-type-hinting-without-cyclic-imports